core/src/main/scala/cats/Alternative.scala

```diff
@@ -89,6 +88,16 @@ import scala.annotation.implicitNotFound
     }
 }
 
+private[cats] trait AlternativeBinCompat0[F[_]] { self: Alternative[F] =>
+
+  def unite[G[_], A](fga: F[G[A]])(implicit FM: FlatMap[F], G: Foldable[G]): F[A] =
+    unite0(fga)
+
+  // Allows to share implementation between the old and new `unite` method versions.
+  final protected def unite0[G[_], A](fga: F[G[A]])(implicit FM: FlatMap[F], G: Foldable[G]): F[A] =
+    FM.flatMap(fga) { G.foldMapK(_)(pure)(self) }
```

Comment:
I tried to change implementation from
```
FM.flatMap { a =>
  G.foldLeft(ga, empty[A])((acc, a) => combineK(acc, pure(a)))
}
```
to just
```
FM.flatMap(fga) { G.foldMapK(_)(pure)(self) }
```
It should be equivalent to the previous one. But can be even more efficient since `foldMapK` has optimized implementations at least for some of supported collections.

Reply:
... but if there're any concerns or objections with this approach, I won't resist to switch back to the original implementation.